tensorflow/compiler/xla/service/pattern_matcher_test.cc

```diff
@@ -1102,5 +1102,94 @@ TEST_F(PatternMatcherTest, CustomCallMatchers) {
       root, m::CustomCall("test_target", m::Parameter(1), m::Parameter(0))));
 }
 
+TEST_F(PatternMatcherTest, OptionalUnaryOp) {
+  constexpr char kModuleStr[] = R"(
+    HloModule test_module
+
+    ENTRY test {
+      p0 = f32[] parameter(0)
+      cos = cosine(p0)
+      ROOT out = f32[] abs(cos)
+    }
+  )";
+  TF_ASSERT_OK_AND_ASSIGN(auto hlo_module,
+                          ParseAndReturnVerifiedModule(kModuleStr));
+  auto* root = hlo_module->entry_computation()->root_instruction();
+
+  EXPECT_TRUE(Match(
+      root,
+      m::OptionalUnaryOp({HloOpcode::kBitcast, HloOpcode::kCos}, m::Abs())));
+  EXPECT_TRUE(
+      Match(root, m::OptionalUnaryOp({HloOpcode::kBitcast, HloOpcode::kCos},
+                                     m::Abs(m::Cos()))));
+  EXPECT_TRUE(Match(
+      root, m::OptionalUnaryOp({HloOpcode::kCos, HloOpcode::kAbs}, m::Cos())));
+  EXPECT_FALSE(Match(
+      root,
+      m::OptionalUnaryOp({HloOpcode::kCos, HloOpcode::kAbs}, m::Bitcast())));
+  EXPECT_FALSE(Match(
+      root,
+      m::OptionalUnaryOp({HloOpcode::kCos, HloOpcode::kBitcast}, m::Cos())));
+
+  std::string description = absl::StrCat(
+      "an HloInstruction which optionally matches a unary operand with one of "
+      "the opcodes {",
+      HloOpcodeString(HloOpcode::kCos), ", ", HloOpcodeString(HloOpcode::kAbs),
+      "} before matching the given pattern, an HloInstruction with opcode ",
```

Comment:
s/the given pattern,//

Reasoning: Everything in this string is a pattern that was given somewhere.  So "the given pattern" doesn't add extra information.  Moreover users may look at it and try to figure out how it *does* add extra information, which is confusing.